R/compute.R

```diff
@@ -309,29 +309,43 @@ get_aks_compute_credentials <- function(cluster) {
 #' If you want to secure your AKS cluster using an Azure Virtual Network, you
 #' must create the virtual network first. For more information, see
 #' [Secure Azure ML experimentation and inference jobs within an Azure Virtual Network](https://docs.microsoft.com/en-us/azure/machine-learning/service/how-to-enable-virtual-network#aksvnet)
+#'
+#' If you want to re-attach an AKS cluster, for example to to change SSL or other
+#' cluster configuration settings, you must first remove the existing attachment
+#' with `detach_aks_compute()`.
+#'
+#' Attaching a cluster will take approximately 5 minutes.
 #' @param workspace The `Workspace` object to attach the AKS cluster to.
-#' @param cluster_name A string of the name for the cluster.
-#' @param resource_id A string of the resource ID for the AKS cluster being
-#' attached.
 #' @param resource_group A string of the resource group in which the AKS cluster
 #' is located.
+#' @param cluster_name A string of the name of the AKS cluster.
+#' @param cluster_purpose The targeted usage of the cluster. The possible values are
+#' "DevTest" or "FastProd". This is used to provision Azure Machine Learning components
+#' to ensure the desired level of fault-tolerance and QoS. If your cluster has less
+#' than 12 virtual CPUs, you will need to specify "DevTest" for this argument. We
+#' recommend that your cluster have at least 2 virtual CPUs for dev/test usage.
 #' @return The `AksCompute` object.
 #' @export
-#' @examples
-#' \dontrun{
+#' @section Examples:
```

Comment:
why "@section Examples:" instead of "@examples"?

Reply:
@examples will render all the example code inside an if (FALSE) { ...} block in the pkgdown docs site since all of our example code is marked dontrun: e.g. https://azure.github.io/azureml-sdk-for-r/reference/create_workspace.html

we originally switched from @section Examples: to @examples so we could pass the first CRAN release (one of the reviewers had a nit about this), and one of my open work items was to switch this back after we successfully published to CRAN. Since none of our example code chunks can be independently run (since we need the workspace context, etc.) all of the code will have the if (FALSE) {} around it which ends up looking strange in the docs website. What do you think?

Reply:
ah got it - sounds good. We should have a work item to to eventually clean this up in all of the examples.